code/datums/helper_datums/teleport.dm

```diff
@@ -170,45 +170,46 @@
 	return 1
 
 /datum/teleport/instant/science/teleportChecks()
-	if(istype(teleatom, /obj/item/weapon/disk/nuclear)) // Don't let nuke disks get teleported --NeoFite
-		teleatom.visible_message("<span class='danger'>The [teleatom] bounces off of the portal!</span>")
-		return 0
-	if(teleatom.locked_to)
-		return 0
-
-	if(!isemptylist(teleatom.search_contents_for(/obj/item/weapon/disk/nuclear)))
-		if(istype(teleatom, /mob/living))
-			var/mob/living/MM = teleatom
-			MM.visible_message("<span class='danger'>The [MM] bounces off of the portal!</span>","<span class='warning'>Something you are carrying seems to be unable to pass through the portal. Better drop it if you want to go through.</span>")
-		else
-			teleatom.visible_message("<span class='danger'>The [teleatom] bounces off of the portal!</span>")
-		return 0
-
-	if(destination.z == map.zCentcomm) //centcomm z-level
-		if(istype(teleatom, /obj/mecha) && (universe.name != "Supermatter Cascade"))
-			var/obj/mecha/MM = teleatom
-			to_chat(MM.occupant, "<span class='danger'>The mech would not survive the jump to a location so far away!</span>")//seriously though, why? who wrote that?
-
+	for (var/atom/movable/AM in recursive_type_check(teleatom))
```

Comment:
Oh shit I didn't notice that HALF the code already did recursive checks and the other half didn't.